lib/build.py

```diff
@@ -265,6 +266,19 @@ def execute(self):
    def execute_(self):
       ...
 
+   def metadata_history_add(self, inst_ct):
+      hist = { "history": [ { "created": ch.now_utc_iso8601(),
+                              "created_by": "%s %s" % (self.str_name(), self.str_())}]}
+      images[image_i].metadata_append(hist)
+
+   def metadata_history_remove(self):
+      images[image_i].metadata_history_remove()
+
+   def metadata_history_update(self, inst_ct):
+     hist = images[image_i].metadata_history_return()
+     images[image_i].metadata_history_replace(hist)
+     self.metadata_history_add(inst_ct)
+
```

Comment:
These can probably be eliminated in favor of calling the `Image` methods directly.